modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceVirtualProducts.testcase

```diff
@@ -15,6 +15,141 @@ definition {
 		CommerceConfiguration.commerceTearDown();
 	}
 
+	@description = "This is a test case for COMMERCE-5582. A new virtual product with no saved configuration will use the default settings and check out with no console errors. The download link of the virtual product is only displayed when the configuration is saved"
+	@priority = 3
+	test CanCheckoutVirtualProductWithNoSavedConfiguration {
+		property portal.acceptance = "false";
+
+		task ("Given a Minium site with an account is created") {
+			CommerceAccelerators.initializeNewSiteViaAccelerator(siteName = "Minium");
+
+			Navigator.openWithAppendToBaseURL(urlAppend = "web/minium");
+
+			CommerceNavigator.gotoMiniumSidebarItem(menuItem = "Account Management");
+
+			CommerceAcceleratorsInitializer.createNewAccountViaFront(accountName = "Test Account");
+		}
+
+		task ("And a virtual product is created") {
+			CommerceJSONProductsAPI._addCommerceProduct(
+				catalogName = "Minium",
+				gtin = "GTIN1",
+				manufacturerPartNumber = "MPN1",
+				productDescription = "Full Description",
+				productName = "Virtual Product",
+				productType = "Virtual",
+				shortDescription = "Short Description",
+				sku = "VIRTUALSKU");
+		}
+
+		task ("And the admin selects a file for the virtual product along with the activation status 'Pending' but doesn't save") {
+			CommerceProducts.openProductsAdmin();
+
+			CommerceNavigator.searchEntry(entryName = "Virtual");
+
+			CommerceNavigator.gotoEntry(entryName = "Virtual Product");
+
+			CommerceEntry.gotoMenuTab(menuTab = "Virtual");
+
+			Click(locator1 = "FormFields#SELECT_UPLOAD_FIELD");
+
+			SelectFrameTop();
+
+			SelectFrame(
+				key_frameTitle = "Select File",
+				locator1 = "CommerceEntry#ANY_IFRAME");
+
+			UploadCommonFile.uploadCommonFileHiddenNoMouseOver(
+				locator1 = "TextInput#FILE",
+				value1 = "Commerce_Black.jpg");
+
+			Click(
+				key_filterOption = "Add",
+				locator1 = "AppBuilder#CHOOSE_SEARCH_FILTER_BY_OPTION");
+
+			SelectFrameTop();
+
+			WaitForElementNotPresent(locator1 = "CommerceEntry#ANY_IFRAME");
+
+			if (IsElementPresent(key_contentType = "Base Information", locator1 = "CommerceEntry#NOTIFICATION_DELIVERY_PANEL_CLOSED")) {
+				Click(
+					key_contentType = "Base Information",
+					locator1 = "CommerceEntry#NOTIFICATION_DELIVERY_PANEL_CLOSED");
+			}
+
+			Select(
+				key_fieldLabel = "Activation Status",
+				locator1 = "Select#GENERIC_SELECT_FIELD",
+				value1 = "Pending");
+		}
```

Comment:
Can you remove all these steps? For this issue, we don't need to configure anything. We just need to create the product

Thanks!